### ISSUE TITLE
Incorrect albumname is used when refreshing album information

### ISSUE THREAD

**Issue by ronie** [xbmc/xbmc#18036]
## Bug report
When manually refreshing an album, Kodi passes the previously scraped albumname to the scraper instead of the albumname retrieved from the tags.

### Describe the bug
Example: during an initial library scan, the scraper picked the wrong result for the album 
`Gabrielle - Rise` (it selected the EP instead of the Album).
This EP was added as `Gabrielle - Rise (Exclusive 4 Track Sampler)` to my database.

When i manually refresh this album, Kodi ask the scraper to search for `Gabrielle - Rise (Exclusive 4 Track Sampler)`.
Now MusicBrainz will offer only 1 result (the EP). So it's not possible to correct the mistake and retrieve the album instead.

## Expected Behavior
When manually refreshing it should use the albumname from the tags, just like it does with the initial library scan.

## Actual Behavior
It uses the previously scraped albumname.

### Debuglog
https://paste.kodi.tv/irutayizel.kodi
lines 815-817 - initial scrape
lines 999-1001 - manual refresh

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Manjaro Linux
 - Kodi version: v19 (compiled 2020-06-06)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ronie** [comment]
edit: this bug report is not related to https://github.com/xbmc/xbmc/pull/18035, which fixes a different issue with manual refresh.

**Comment by DaveTBlake** [comment]
I see you are having fun with album scraping @ronie , OK have it on my list

**Comment by DaveTBlake** [comment]
@ronie can you confirm if you had the "Prefer online information" setting enabled when you ran the scraper? I think you probably did, and consequently the scraped album title overwrote the value that had been read from tags and it is no longer available to use in any refresh or rescraping. Should it go back to rescan the embedded metadata, or alternatively simply not have an option to overwrite album title?  My choice would be to replace the "Prefer online information" setting, and control how  artist or album data that gets initially set from tag data can be overwriten by scraping in a more detailed way.

**Comment by ronie** [comment]
thanx for looking into it, i indeed have 'prefer online information' enabled.

i had another look at the core code and i think the solution is pretty simple here,
when refreshing, we should also display the selection dialog if there's only 1 result.

the selection dialog offers the user to manually correct the artist name / album title
![select](https://user-images.githubusercontent.com/687265/84261893-e5240b00-ab1c-11ea-82fd-ce1bd3a31772.jpg)

it's an easy fix (https://github.com/xbmc/xbmc/pull/18035/commits/c07fe7bad68344aa8e48eb0d9072b81fbf450c7c) which i've now added to https://github.com/xbmc/xbmc/pull/18035

**Comment by DaveTBlake** [comment]
Hum, won't that mean always seeing the select dialog after hitting the refresh button? Not sure we want that, especially if just refreshing to get new list of possible art for an album that was tagged with mbids so is definitely known and not to be replaced with some other

**Comment by ronie** [comment]
> Hum, won't that mean always seeing the select dialog after hitting the refresh button?

nope, you won't get the select dialog if you're files are tagged with mbid's.

when a mbid from tags is available kodi will ask the scraper to provide the albumdetails for the given mbid. it does not ask to scraper to search for possible matched bases on artistname/albumtitle in that case.
